internal/server/server.go

```diff
@@ -79,6 +85,15 @@ func (srv *Server) Provision(_ context.Context, req *pb.ProvisionRequest) (*pb.P
 		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("error parsing Wireguard key: %s", err))
 	}
 
+	var grpcPeerAddrPort string
+
+	if req.WireguardOverGrpc != nil && *req.WireguardOverGrpc {
+		grpcPeerAddrPort, err = srv.cfg.PeerProvider.PeerAddrPort(pubKey)
+		if err != nil {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("error generating tunnel endpoint: %s", err))
+		}
+	}
+
 	srv.eventCh <- wireguard.PeerEvent{
 		PubKey:                      pubKey,
```

Comment:
PeerEvent should contain the allocated tunnel address, as it needs to be persisted, same way this code should restore the tunnel config via `PeerEvent`

Reply:
Done!